networkx/generators/community.py

```diff
@@ -725,9 +697,34 @@ def _powerlaw_sequence(gamma, low, high, condition, length, max_iters, seed):
     raise nx.ExceededMaxIterations("Could not create power law sequence")
 
 
-# TODO Needs documentation.
+def _hurwitz_zeta(x, q, tolerance):
+    """The Hurwitz zeta function, or the Riemann zeta function of two arguments.
+
+    ``x`` must be greater than one and ``q`` must be positive.
+
+    This function repeatedly computes subsequent partial sums until
+    convergence, as decided by ``tolerance``.
+    """
+    z = 0
+    z_prev = -float("inf")
+    k = 0
+    while abs(z - z_prev) > tolerance:
+        z_prev = z
+        z += 1 / ((k + q) ** x)
+        k += 1
+    return z
+
+
 def _generate_min_degree(gamma, average_degree, max_degree, tolerance, max_iters):
     """Returns a minimum degree from the given average degree."""
+    # Defines zeta function whether or not Scipy is available
+    try:
+        from scipy.special import zeta
+    except ImportError:
+
+        def zeta(x, q):
```

Comment:
Should there be a warning if scipy isn't found? I imagine the pure python implementation will be much slower, and a weird slowdown may be difficult to debug if the user accidentally uninstalls scipy.

Reply:
It's no different than the existing code. Warnings for missing scipy could be addressed in a separate PR if that's an issue.